lsp/src/index_engine.rs

```diff
@@ -75,7 +78,19 @@ impl IndexEngine {
             state.publish_diagnostics(*id, diags).await;
         }
 
-        state.client.inlay_hint_refresh().await.unwrap();
+        if state
+            .client_capabilities
+            .workspace
+            .as_ref()
+            .is_some_and(|w| {
+                w.inlay_hint
+                    .as_ref()
+                    .is_some_and(|i| i.refresh_support.is_some_and(|r| r))
+            })
+        {
```

Comment:
Another way to do this.

```suggestion
       if matches!(
            &state.client_capabilities.workspace,
            Some(WorkspaceClientCapabilities {
                inlay_hint: Some(InlayHintWorkspaceClientCapabilities {
                    refresh_support: Some(_),
                    ..
                }),
                ..
            })
        ) {
```